aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactory.java

```diff
@@ -315,4 +302,81 @@ private static Object appSyncOpValue(QueryOperator<?> qOp) throws DataStoreExcep
                     amplifyException, "Validate your model file.");
         }
     }
+
+    private static Map<String, Object> getMapOfFieldNameAndValues(
```

Comment:
What's the reasoning for moving this out of `ModelSchema`?    And why is the implementation of this method not identical to the similar method created in `AppSyncGraphQLRequestFactory`?

Ultimately, I think `AppSyncRequestFactory` (used for DataStore) and `AppSyncGraphQLRequestFactory` (used for API) should be combined, or at least have a common superclass maybe.

Reply:
Maybe have some of the common functionality could go in the `aws-api-appsync` package since it's used by both api and datastore?

Reply:
These are good points, thanks guys. The original point of the `aws-api-appync` package was to contain AppSync-specific code. I agree that we should probably have just one of these, and it should live in that module. Will make this change.

Reply:
That doesn't _have_ to be in this PR if you wanted to split it out.  But, go for it if you want!

I was more interested in the reasoning for moving this method out of `ModelSchema`?

Reply:
The two request factories have little differences laced throughout - the response type, whether to include _version, the request options object, etc. … :cry:

Long term, I would like to see all of the AppSync-specific stuff go into `aws-api-appsync`; I still think that's a good direction.

I had moved the request formation logic out of the `ModelSchema` since `ModelSchema` lives in core. I didn’t want core doesn’t know about `SerializedModel`s. This `getMapOfFieldNameAndValues(...)` method really _is_ about forming an outbound request, I think, not something having to do with the schema in-and-of-itself.